### ISSUE TITLE
Background Tasks

### ISSUE THREAD

**Issue by kryptur** [MyTFG/mytfg-server-7#2]
Sollte der Server quasi "dauerhaft" laufen und eine Warteschlange (z.B. für das Erstellen von Nutzern) haben oder eher so wie ein Skript alles durchlaufen und sich dann beenden?

Für den ersten Ansatz wären dann verschiedene Threads nötig, welche die API Request annehmen, ggfs. auf andere Threads aufteilen, einen "Queue Thread", der alle Möglichen Aufgaben abarbeitet usw. 
Könnte interessant zu implementieren sein - hat aber bestimmt auch Vorteile.

Meinungen?

**Comment by rleh** [comment]
Auf jeden Fall letzteres.
Ersteres spielt doch die Vorteile gegenüber dem aktuellen (skriptbasiertem) System gar nicht aus.

ich weiß ja nicht wie viel du da selbst implementieren willst, aber ich glaube man kann sich da viel Arbeit abnehmen lassen...  [JAX-RS](https://en.wikipedia.org/wiki/Java_API_for_RESTful_Web_Services) ([Apache CFX](https://cxf.apache.org/)/[Jersey](https://jersey.java.net/))

**Comment by kryptur** [comment]
Du meinst also eigentlich die erste Variante, nicht die letztere?! 

Also selbst implementieren hätte ich jetzt gedacht, dass die Threads für gewisse Tasks gestartet werden usw. Den Https Proxy natürlich nicht. Im Grunde nur, dass es eine zentrale Server Instanz gibt, die läuft. Und zwar unabhängig davon, ob ne anfrage kommt oder nicht. Der kann dann eigenständig cronjobs ausführen, im Hintergrund irgendwelche liegen gebliebenen Sachen machen, Statistiken erstellen, Mails versenden usw.

Was davon würdest du denn da auslagern?

**Comment by rleh** [comment]
Ja, ich meinte die erste Variante :smiley:, hab dem Kommentar oben mal editiert.

Ja, klingt für mich sehr sinnvoll was du machen willst.  Aber ich hätte keine Ahnung wie ich das in Java machen würde, daher hätte ich mich da einfach z.B. an ein Jersey Example gehalten und damit Jersey verwendet. Was Jersey dann da genau tut hab ich allerdings noch nicht raus :smile:

**Comment by kryptur** [comment]
Ok :smile: 

Naja, also Threads sind recht einfach, da brauchen wir eigentlich nix externes zu...
